### PR TITLE
Make error message when component failed to initialize more helpful

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Frameworks Design member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

File: lib/src/component_declaration/component_base.dart [hunk 1]
```diff
@@ -574,13 +574,13 @@ abstract class UiProps extends MapBase
     // down the line instead.
     assert(
         componentFactory != null,
-        'componentFactory is null. Possible causes:\n'
-        '1. Something went wrong when initializing the `\$${runtimeType}Factory` variable in the generated code.\n'
-        '   Check for errors in the console for more information.'
-        '2. This is a props map view factory (declared as just a factory and props), and should not be invoked.\n'
+        'Most likely, another error occurred while initializing this component\'s declaration (via `\$${runtimeType}Factory`); '
+        'to find the real issue, look for earlier errors in theh console output.\n\n'
```
Comment (greglittlefield-wf) [inline lib/src/component_declaration/component_base.dart:578]:
```suggestion
        'to find the real issue, look for earlier errors in the console output.\n\n'
```

Fix typo

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by joebingham-wk — APPROVED** [review]
+10

**Comment by greglittlefield-wf** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**